internal/nozzle/nozzle.go

```diff
@@ -239,31 +241,47 @@ func (d *Nozzle) PostMetrics() {
 	d.ResetSlowConsumerError()
 }
 
-func (d *Nozzle) handleError(err error) {
+func (d *Nozzle) handleError(err error) bool {
+	// If error is a retry error, we log it and let the consumer retry.
 	if retryErr, ok := err.(noaaerrors.RetryError); ok {
-		err = retryErr.Err
+		d.log.Errorf("Error while reading from the firehose: %v", retryErr.Error())
+		d.log.Info("The Firehose consumer hit a retry error, retrying ...")
+		//TODO: Why should we alert with a metric? like for `AlertSlowConsumerError`?
+		return true
```

Comment:
You can't return here, otherwise you will never enter the switch. And you want to because `websocket.CloseError` are of type `noaaerrors.RetryError`

Reply:
Yeap found this one as well. Nice!